src/web/rustdoc.rs

```diff
@@ -46,21 +47,12 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
         permanent: bool,
         path_in_crate: Option<&str>,
     ) -> IronResult<Response> {
-        let mut queries: std::collections::BTreeMap<
-            std::borrow::Cow<'_, str>,
-            std::borrow::Cow<'_, str>,
-        > = std::collections::BTreeMap::new();
+        let mut queries = HashMap::new();
```

Comment:
with a hashmap the ordering is not guaranteed, as such the params can appear in a different order

Reply:
yeah, I discovered that too in the failing tests :) 

(